woocommerce/woocommerce-gutenberg-products-block
# Create Context Provider for Notices w/ Notices API

This PR implements a system for creating and outputting notices for a given context.

Fixes #1795

**Usage**

Wrap your components in a `StoreNoticesProvider` component and give it a custom context.

```jsx
import StoreNoticesProvider from '@woocommerce/base-context/store-notices-context';
```

```jsx
<StoreNoticesProvider context="wc/cart">
		{children}
</StoreNoticesProvider>
```

Too create notices, use the `useStoreNotices` hook. Example:

```jsx
const {
		addSuccessNotice
} = useStoreNotices();

addSuccessNotice( "Well done!" );
```

This hook providers a series of methods for notice manipulation under the current context:

- `addDefaultNotice`
- `addErrorNotice`
- `addWarningNotice`
- `addInfoNotice`
- `addSuccessNotice` 
- `removeNotices`

And finally, notices are rendered in a container using the Notice component (https://github.com/WordPress/gutenberg/tree/master/packages/components/src/notice) -- these are given woocommerce class names for styling.

### Screenshots

![Screenshot 2020-03-02 at 14 17 45](https://user-images.githubusercontent.com/90977/75684389-9bf82b80-5c90-11ea-939b-c2a4a2f7c4db.png)

### How to test the changes in this Pull Request:

1. Go to the cart.
2. Apply a coupon - see message.
3. Apply it again. See error.
4. Remove the coupon. See info message.
